### PR TITLE
Always retain `fixedType`, add unit test which fails if not

### PR DESCRIPTION
**Pull request by madisonchamberlain**
### Description

Fallback to retaining `fixedType` chunk if its not one of the types in the if/if else statements. Add a test which ensures that this works. 

### Checklist
- [X] Created tests which fail without the change (if possible)

### CODE REVIEW

File: chunk_test.go [hunk 1]
```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
...
+func TestRetainChunkWOHighPrecision(t *testing.T) {
+	runDBTest(t, func(dbt *DBTest) {
+		query := "select 0"
+
+		ctx := WithArrowBatches(context.Background())
+
+		var rows driver.Rows
+		var err error
+
+		// must use conn.Raw so we can get back driver rows (an interface)
```
Comment (sfc-gh-pfus) [inline chunk_test.go:649]: In this repo we know this, we don't need to put this comment :)

File: chunk_test.go [hunk 2]
```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
...
+		// which can be cast to snowflakeRows which exposes GetArrowBatch
+		err = dbt.conn.Raw(func(x interface{}) error {
+			queryer, implementsQueryContext := x.(driver.QueryerContext)
+			assertTrueF(t, implementsQueryContext, "snowflake connection driver does not implement queryerContext")
```
Comment (sfc-gh-pfus) [inline chunk_test.go:653]: Same here, this is fine to do casting without this second parameter. Let it burn otherwise!

File: chunk_test.go [hunk 3]
```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
...
+
+			rows, err = queryer.QueryContext(WithArrowBatches(ctx), query, nil)
```
Comment (sfc-gh-pfus) [inline chunk_test.go:655]: This context is already arrow batches aware, right?

File: chunk_test.go [hunk 4]
```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
...
+			return err
+		})
+
+		assertNilF(t, err, "error running select query")
+
+		sfRows, isSfRows := rows.(SnowflakeRows)
```
Comment (sfc-gh-pfus) [inline chunk_test.go:661]: Same here, assume casting succeeds.

File: chunk_test.go [hunk 5]
```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
...
+
+		arrowBatches, err := sfRows.GetArrowBatches()
+		assertNilF(t, err, "error getting arrow batches")
+		assertEqualF(t, len(arrowBatches), 1, "should have one batch")
+
+		records, err := arrowBatches[0].Fetch()
+		assertNilF(t, err, "error getting batch")
+		assertNotNilF(t, records, "records should not be nil")
+
+		recs := *records
```
Comment (sfc-gh-pfus) [inline chunk_test.go:672]: Do we need this additional variable?

File: chunk_test.go [hunk 6]
```diff
@@ -636,3 +636,51 @@ func TestQueryArrowStreamDescribeOnly(t *testing.T) {
...
+		numRecords := len(recs)
+		assertEqualF(t, numRecords, 1, "should have exactly one record")
+
+		record := recs[0]
+		assertEqualF(t, len(record.Columns()), 1, "should have exactly one column")
+
+		column := record.Column(0)
```
Comment (sfc-gh-pfus) [inline chunk_test.go:679]: Maybe we can have `record.Column(0).(*array.Int8)`?

File: chunk_test.go (hunk 1 above)
Comment (madisonchamberlain) [reply chunk_test.go]: true, deleted!

File: chunk_test.go (hunk 3 above)
Comment (madisonchamberlain) [reply chunk_test.go]: Yes, updated

File: chunk_test.go (hunk 4 above)
Comment (madisonchamberlain) [reply chunk_test.go]: done!

File: chunk_test.go (hunk 5 above)
Comment (madisonchamberlain) [reply chunk_test.go]: Nope, deleted

File: chunk_test.go (hunk 6 above)
Comment (madisonchamberlain) [reply chunk_test.go]: good idea, updated

**Review by sfc-gh-pfus — APPROVED** [review]

**Pull request merged**